js/main.js

```diff
@@ -1,45 +1,534 @@
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 ];
 
-/*----- app's state (variables) -----*/
+let playerShips = {
+  aircraftCarrier: {
+    name: "Aircraft Carrier",
+    length: 5,
+    health: 5,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "player-ac",
+  },
+  battleship: {
+    name: "Battleship",
+    length: 4,
+    health: 4,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "player-bs",
+  },
+  cruiser: {
+    name: "cruiser",
+    length: 3,
+    health: 3,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "player-cr",
+  },
+  firstDestroyer: {
+    name: "First Destroyer",
+    length: 3,
+    health: 3,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "player-fd",
+  },
+  secondDestroyer: {
+    name: "Second Destroyer",
+    length: 3,
+    health: 3,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "player-sd",
+  },
+  firstSubmarine: {
+    name: "First Submarine",
+    length: 2,
+    health: 2,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "player-fs",
+  },
+  secondSubmarine: {
+    name: "Second Submarine",
+    length: 2,
+    health: 2,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "player-ss",
+  },
+};
+let cpuShips = {
+  aircraftCarrier: {
+    length: 5,
+    health: 5,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "cpu-ac",
+  },
+  battleship: {
+    length: 4,
+    health: 4,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "cpu-bs",
+  },
+  cruiser: {
+    length: 3,
+    health: 3,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "cpu-cr",
+  },
+  firstDestroyer: {
+    length: 3,
+    health: 3,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "cpu-fd",
+  },
+  secondDestroyer: {
+    length: 3,
+    health: 3,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "cpu-sd",
+  },
+  firstSubmarine: {
+    length: 2,
+    health: 2,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "cpu-fs",
+  },
+  secondSubmarine: {
+    length: 2,
+    health: 2,
+    firstIndex: undefined,
+    orientation: undefined,
+    key: "cpu-ss",
+  },
+};
+const BASE_PLAYER_SHIP_STATES = JSON.parse(JSON.stringify(playerShips));
+const BASE_CPU_SHIP_STATES = JSON.parse(JSON.stringify(cpuShips));
+const NUM_OF_SHIPS = Object.keys(playerShips).length - 1;
+const LENGTH = CPU_BOARD[0].length;
+const HEIGHT = CPU_BOARD.length;
+const PLAYER_DIVS = [];
+const CPU_DIVS = [];
+const LOCATIONS = new Set();
+const playAgainBtn = document.createElement("button");
+const PLAYER = 1;
+const CPU = -1;
+const ACTIVE_SHIP = 1;
```

Comment:
Now I'm being nitpicky but it might make a bit more sense to put these in an object called `BOARD_STATES` or something. Just to give some more context, and make it clear that they're related. i.e.

```
const BOARD_STATES = {
    ACTIVE_SHIP: 1,
    HIT_SHIP: 2,
    ...etc
}
```